pkg/pulumiyaml/packages.go

```diff
@@ -397,8 +397,15 @@ func (p resourcePackage) FunctionTypeHint(typeName FunctionTypeToken) InputTypeH
 	if !ok {
 		return nil
 	}
+	hints := []*schema.Property{}
+	if f.Outputs != nil {
+		hints = append(hints, f.Outputs.Properties...)
+	}
+	if f.Inputs != nil {
+		hints = append(hints, f.Inputs.Properties...)
+	}
 	return inputTypeHint{
-		TypeHint:   fieldTypeHint{append(f.Outputs.Properties, f.Inputs.Properties...)},
+		TypeHint:   fieldTypeHint{hints},
```

Comment:
this coincidentally also fixes a potentially sneaky data race later - `append()` modifies the backing store of the first argument